api-2.6/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationDispenseTranslatorImpl_2_6.java

```diff
@@ -196,22 +210,38 @@ public MedicationDispense toOpenmrsType(@Nonnull MedicationDispense openmrsObjec
 			MedicationDispensePerformerComponent performerComponent = fhirObject.getPerformerFirstRep();
 			if (performerComponent != null && performerComponent.hasActor()) {
 				openmrsObject.setDispenser(practitionerReferenceTranslator.toOpenmrsType(performerComponent.getActor()));
+			} else {
+				openmrsObject.setDispenser(null);
 			}
+		} else {
+			openmrsObject.setDispenser(null);
 		}
 		
 		if (fhirObject.hasSubstitution()) {
 			MedicationDispenseSubstitutionComponent substitution = fhirObject.getSubstitution();
 			openmrsObject.setWasSubstituted(substitution.getWasSubstituted());
 			if (substitution.hasType()) {
 				openmrsObject.setSubstitutionType(conceptTranslator.toOpenmrsType(substitution.getType()));
+			} else {
+				openmrsObject.setSubstitutionType(null);
 			}
 			if (substitution.hasReason()) {
 				openmrsObject.setSubstitutionReason(conceptTranslator.toOpenmrsType(substitution.getReasonFirstRep()));
+			} else {
+				openmrsObject.setSubstitutionReason(null);
 			}
+		} else {
+			openmrsObject.setWasSubstituted(null);
+			openmrsObject.setSubstitutionType(null);
+			openmrsObject.setSubstitutionReason(null);
 		}
 		
-		if (fhirObject.getMeta() != null) {
-			openmrsObject.setDateCreated(fhirObject.getMeta().getLastUpdated());
+		if (fhirObject.getMeta() != null && fhirObject.getMeta().getLastUpdated() != null) {
+			if (openmrsObject.getDateCreated() == null) {
+				openmrsObject.setDateCreated(fhirObject.getMeta().getLastUpdated());
+			} else {
+				openmrsObject.setDateChanged(fhirObject.getMeta().getLastUpdated());
+			}
```

Comment:
This is tangential to the ticket, but seemed like the correct behavior... though, perhaps it begs the question, should we really be updating these OpenMRS "audit" fields at all?

Reply:
Merging this in, but still curious about thoughts as to whether we should up updated OpenMRS audit fields here... @ibacher  @mseaton

Reply:
Oooh... I missed this. I actually don't like the idea of manually updating the audit fields at all... IMO, we can use them to display values when it makes sense, but I don't love allowing anyone to actually edit them.

Reply:
thoughts @mseaton?

Reply:
Fine with me.